feat(cli): silence noise + add helpful error for missing references

kennyderek opened the pull request:
tested with Square and SPS.

We don't have a 1000+ line laundry list of warnings now. More helpful errors are surfaced via fern check

<img width="1711" height="861" alt="image" src="https://github.com/user-attachments/assets/af3be25e-b8c8-4858-8bf8-a1c7bc58c955" />

kennyderek committed:
silence noise + add helpful error

dsinghvi reviewed (approved):
@kennyderek can you include an entry in versions.yml as well so we can release for the CLI

kennyderek committed:
version.yml

devin-ai-integration[bot] committed:
fix(cli): fix biome lint errors and compile error

- Replace 'any' types with 'unknown' in valid-local-references.ts
- Fix non-null assertion by using proper null check
- Fix DefinitionFile import path in listConjureFiles.ts

Co-Authored-By: kenny@buildwithfern.com <kenny@buildwithfern.com>

devin-ai-integration[bot] committed:
fix(cli): fix TypeReferenceDetailedSchema compile error

- Rewrite type guard to use Exclude instead of missing export
- Avoid depending on TypeReferenceDetailedSchema export from schema package

Co-Authored-By: kenny@buildwithfern.com <kenny@buildwithfern.com>

kennyderek committed:
Merge branch 'main' into kenny/silence-noisy-errors
